geode-core/src/test/java/org/apache/geode/internal/cache/GemFireCacheImplTest.java

```diff
@@ -620,6 +625,40 @@ public void getCacheServers_isCanonical() {
         .isSameAs(gemFireCacheImpl.getCacheServers());
   }
 
+  @Test
+  public void testLockDiskStore() throws InterruptedException {
+    int nThread = 10;
+    String diskStoreName = "MyDiskStore";
+    AtomicInteger nTrue = new AtomicInteger();
+    AtomicInteger nFalse = new AtomicInteger();
+    ExecutorService executorService = Executors.newFixedThreadPool(nThread);
+    IntStream.range(0, nThread).forEach(tid -> {
+      executorService.submit(() -> {
+        try {
+          boolean lockResult = gemFireCacheImpl.doLockDiskStore(diskStoreName);
+          if (lockResult) {
+            nTrue.incrementAndGet();
+          } else {
+            nFalse.incrementAndGet();
+          }
+        } finally {
+          boolean unlockResult = gemFireCacheImpl.doUnlockDiskStore(diskStoreName);
+          if (unlockResult) {
+            nTrue.incrementAndGet();
+          } else {
+            nFalse.incrementAndGet();
+          }
+        }
+      });
+    });
+    executorService.shutdown();
+    executorService.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
```

Comment:
If you use the Rule, you shouldn't need this or the shutdown. In general, you should however use `GeodeAwaitility.getTimeout()` instead of `Long.MAX_VALUE`.

Reply:
Will change it.

Reply:
Done.